### PR TITLE
Add tool execution layer foundation (Phase 1 of MCP support)

### PR DESCRIPTION
**Pull request by xerial**
## Summary

This PR implements Phase 1 of the Model Context Protocol (MCP) support by adding the foundational tool execution layer to scala-ai. This enables AI agents to execute tools automatically during chat sessions.

### Core Components

1. **ToolExecutor trait** (`ai-agent/src/main/scala/wvlet/ai/agent/tool/ToolExecutor.scala`)
   - Abstract interface for tool execution using `Rx[T]` for async operations
   - Support for executing multiple tools in parallel
   - Tool discovery via `availableTools` and `findTool` methods

2. **LocalToolExecutor** (`ai-agent/src/main/scala/wvlet/ai/agent/tool/LocalToolExecutor.scala`)
   - In-memory implementation for testing and simple tool use cases
   - Function-based tool registry
   - Automatic JSON serialization for non-string results
   - Error handling with proper error messages

3. **ToolEnabledChatSession** (`ai-agent/src/main/scala/wvlet/ai/agent/chat/ToolEnabledChatSession.scala`)
   - Extends ChatSession with automatic tool execution capabilities
   - `chatWithTools` method handles multiple rounds of tool calls
   - Configurable maximum tool execution rounds
   - Extension methods to convert regular ChatSession to tool-enabled

4. **Integration Points**
   - Added `MAX_ROUNDS` to `ChatFinishReason` enum
   - Added `newToolEnabledSession` helper method to `LLMAgent`
   - Extension method `withToolSupport` for ChatSession

## Features

- ✅ Asynchronous tool execution using Airframe's `Rx[T]`
- ✅ Tool registration and discovery
- ✅ Automatic handling of tool calls in chat sessions
- ✅ Support for multiple tool execution rounds
- ✅ Parallel execution of multiple tool calls
- ✅ Comprehensive error handling
- ✅ Full test coverage with AirSpec

## Usage Example

```scala
// Define a tool
val weatherTool = ToolSpec(
  name = "get_weather",
  description = "Get current weather",
  parameters = List(
    ToolParameter("location", "City name", DataType.StringType, None)
  ),
  returnType = DataType.JsonType
)

// Create executor and register tools
val executor = LocalToolExecutor()
  .registerTool(weatherTool, args => {
    Map("temp" -> 25, "condition" -> "sunny", "location" -> args("location"))
  })

// Enable tool execution in chat
val runner = BedrockRunner(agent)
val session = agent.newToolEnabledSession(runner, Some(executor))

// Chat with automatic tool execution
val response = session.chatWithTools("What's the weather in Tokyo?").toSeq.head
// The model calls the tool, gets results, and provides the final answer
```

## Testing

Added comprehensive test suites:
- `LocalToolExecutorTest` - Tests tool registration, execution, error handling
- `ToolEnabledChatSessionTest` - Tests automatic tool execution flow, multiple rounds, edge cases

All tests pass:
```
[info] Passed: Total 12, Failed 0, Errors 0, Passed 12
```

## Next Steps

This foundation enables Phase 2-4 of MCP support:
- Phase 2: MCP client implementation with JSON-RPC protocol
- Phase 3: Tool adaptation between MCP and scala-ai formats
- Phase 4: Integration tests with real MCP servers

## Related Issues

- #159 - Support MCP (Model Context Protocol) tool calling

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: ai-agent/src/main/scala/wvlet/ai/agent/LLMAgent.scala [hunk 1]
```diff
@@ -90,4 +90,21 @@ case class LLMAgent(
...
+    * Create a chat session with tool execution support.
+    *
+    * @param runner
+    *   The agent runner to use
+    * @param toolExecutor
+    *   Optional tool executor for automatic tool execution
+    * @return
+    *   A tool-enabled chat session
+    */
+  def newToolEnabledSession(
```
Comment (xerial) [inline ai-agent/src/main/scala/wvlet/ai/agent/LLMAgent.scala:103]: Using tools is already optional in the parameter, so name this method as `newSession`
